cheatsheets/Cross-Site_Request_Forgery_Prevention_Cheat_Sheet.md

```diff
@@ -1,78 +1,86 @@
-
-# Warning: No Cross-Site Scripting (XSS) Vulnerabilities
-
-[Cross-Site Scripting](https://www.owasp.org/index.php/Cross-Site_Scripting) is not necessary for CSRF to work. However, any cross-site scripting vulnerability can be used to defeat all CSRF mitigation techniques available in the market today (except mitigation techniques that involve user interaction and described later in this cheatsheet). This is because an XSS payload can simply read any page on the site using an XMLHttpRequest (direct DOM access can be done, if on same page) and obtain the generated token from the response, and include that token with a forged request.  This technique is exactly how the [MySpace (Samy) worm](https://en.wikipedia.org/wiki/Samy_%28computer_worm%29) defeated MySpace's anti-CSRF defenses in 2005, which enabled the worm to propagate.
-
-It is imperative that no XSS vulnerabilities are present to ensure that CSRF defenses can't be circumvented. Please see the OWASP [XSS Prevention Cheat Sheet](Cross_Site_Scripting_Prevention_Cheat_Sheet.md) for detailed guidance on how to prevent XSS flaws.
-
-# Resources that need to be protected from CSRF vulnerability
-
-The following list assumes that you are not violating [RFC2616](http://www.w3.org/Protocols/rfc2616/rfc2616-sec9.html#sec9.1.1), section 9.1.1, by using GET requests for state changing operations.
-
-**Note:** If for any reason you violate, you would also need to protect those resources, which is mostly achieved with default `form` `tag` `[GET method]`, `href`, and `src` attributes.
-
-- Form tags with POST
-- Ajax/XHR calls
-
-# CSRF Defense Recommendations Summary
-
-We recommend token based CSRF defense (either stateful/stateless) as a primary defense to mitigate CSRF in your applications. Only for highly sensitive operations, we also recommend a user interaction based protection (either re-authentication/one-time token, detailed in section 6.5) along with token based mitigation.
-
-As a defense-in-depth measure, consider implementing one mitigation from Defense in Depth Mitigations section (you can choose the mitigation that fits your ecosystem considering the issues mentioned under them). These defense-in-depth mitigation techniques are not recommended to be used by themselves (without token based mitigation) for mitigating CSRF in your applications.
-
-# Primary Defense Technique
+# Cross-site request forgery (CSRF) Cheat Sheet
```

Comment:
Would be nice to add:

- More emphasis on using the framework's built in protection (as this should always be the first choice)
- A note about how some methods aren't vulnerable to CSRF (such as Bearer-Token headers)
- Some discussion about how you do CSRF for API calls (if your API accepts session cookies, which many do, how to you protect it?)

Reply:
> Would be nice to add:
> * More emphasis on using the framework's built in protection (as this should always be the first choice)

Done!

> * A note about how some methods aren't vulnerable to CSRF (such as Bearer-Token headers)

Good point!
Cookie based sessions are vulnerable, basic authentication is also vulnerable.
Bearer-Token headers are not. 
Do you have other examples

> * Some discussion about how you do CSRF for API calls (if your API accepts session cookies, which many do, how to you protect it?)

You protect API in the same way as web app - add token to request (e.g. in header) + SameSite + all defence in depth techniques should also work.